### PR TITLE
[BackendUtils]Convert RuntimeBundle to a class

### LINKED ISSUE #1967: Convert RuntimeBundle to object.
RuntimeBundle is currently a struct which contains the information passed forward from compile time to runtime. It contains max allocations, constants, and symbol information. This is a general backend tool so it should be moved to BackendUtils. It should also be made into a class so accessors like getSymbolOffset can be called. The backends will need to be updated as well.

### PR DESCRIPTION
**Pull request by gcatron**
*Description*: Converted RuntimeBundle to a class and added helper methods for getting symbolInfo. Moved RuntimeBundle to BackendUtils. Updated Backends to use new helper methods.
*Testing*: ninja test and ruh.sh both pass
*Documentation*: N/A
Fixes #1967

### CODE REVIEW

File: lib/Backends/CPU/CPUFunction.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 #include "GlowJIT.h"
 
+#include "glow/Backends/BackendUtils.h"
```
Comment (jfix71) [inline lib/Backends/CPU/CPUFunction.h]: Why is it necessary to include here? Can you just include in `CPUFunction.cpp` instead?

File: lib/Backends/Interpreter/InterpreterFunction.h [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #ifndef GLOW_BACKENDS_INTERPRETER_INTERPRETERFUNCTION_H
 #define GLOW_BACKENDS_INTERPRETER_INTERPRETERFUNCTION_H
 
+#include "glow/Backends/BackendUtils.h"
```
Comment (jfix71) [inline lib/Backends/Interpreter/InterpreterFunction.h]: Why is this necessary, does the interpreter use this at all? If so can it be included in the cpp where it's used?

File: lib/Backends/OpenCL/OpenCL.h [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #define GLOW_BACKENDS_OPENCL_OPENCL_H
 
 #include "glow/Backends/Backend.h"
+#include "glow/Backends/BackendUtils.h"
```
Comment (jfix71) [inline lib/Backends/OpenCL/OpenCL.h]: Same as CPU -- can this be moved to `OpenCL.cpp` instead of including in the header?

File: include/glow/Backends/BackendUtils.h [hunk 4]
```diff
@@ -21,6 +21,43 @@
 #include "glow/IR/IR.h"
 
 namespace glow {
+namespace runtime {
+/// RuntimeSymbolInfo
```
Comment (bertmaher) [inline include/glow/Backends/BackendUtils.h]: Nit: I'm not a fan of comments that just repeat the name of the thing being commented :-)

File: include/glow/Backends/BackendUtils.h [hunk 5]
```diff
@@ -21,6 +21,43 @@
...
+/// Contains information for initialization and handling of symbol at runtime.
+struct RuntimeSymbolInfo {
+  /// The size in bytes.
+  size_t size;
+  /// Offset in bytes from the base address.
+  size_t offset;
+  /// Type of symbol.
+  Type type;
+};
+/// Runtime Bundle
```
Comment (bertmaher) [inline include/glow/Backends/BackendUtils.h]: Same nit (name of thing in comment).

File: lib/Backends/CPU/CPUFunction.h (hunk 1 above)
Comment (jfix71) [reply lib/Backends/CPU/CPUFunction.h]: Ah whoops I totally missed what was going on here, ignore me 😀

File: include/glow/Backends/BackendUtils.h (hunk 5 above)
Comment (bertmaher) [reply include/glow/Backends/BackendUtils.h]: I'd also prefer a blank line between top-level class/struct declarations.  (I dunno if this is in the official coding standards but we do it most other places.) (In general I hate blank lines but this is my one exception.) (I also love parenthetical comments.)

File: include/glow/Backends/BackendUtils.h [hunk 6]
```diff
@@ -41,6 +78,7 @@ runtime::RuntimeBundle
 generateRuntimeBundle(const IRFunction &F, MemoryAllocator &constantAllocator,
                       MemoryAllocator &placeholderAllocator,
                       MemoryAllocator &activationsAllocator);
+
```
Comment (bertmaher) [inline include/glow/Backends/BackendUtils.h]: By contrast I could do without this additional blank line ;-).

File: include/glow/Backends/BackendUtils.h [hunk 7]
```diff
@@ -21,6 +21,43 @@
...
+  /// Offset in bytes from the base address.
+  size_t offset;
+  /// Type of symbol.
+  Type type;
+};
+/// Runtime Bundle
+/// Contains the information needed to be passed forward from compile time to
+/// runtime. In order to allocate and initialize memory.
+class RuntimeBundle {
+public:
```
Comment (opti-mix) [inline include/glow/Backends/BackendUtils.h]: Typically, all members of structs are public by default and all members of classes are private by default. And if some internals of the class need to be exposed, it is done via getters and setter. Also the private members would have the "_" suffix in their names.

So, why is `RuntimeBundle` a class, if all its members are public?

File: include/glow/Backends/BackendUtils.h [hunk 8]
```diff
@@ -21,26 +21,65 @@
...
+  /// Amount of memory needed for weights.
+  const size_t constantWeightVarsMemSize_;
+  /// Amount of memory needed for mutable vars.
+  const size_t mutableWeightVarsMemSize_;
+  /// Amount of memory needed for activations.
+  const size_t activationsMemSize_;
+
+public:
+  /// Get Constant Weights memory size.
+  size_t getConstantWeightSize() { return constantWeightVarsMemSize_; }
```
Comment (opti-mix) [inline include/glow/Backends/BackendUtils.h]: it should be a `const` method.

File: include/glow/Backends/BackendUtils.h [hunk 9]
```diff
@@ -21,26 +21,65 @@
...
+  /// Get Mutable Weights memory size.
+  size_t getMutableWeightSize() { return mutableWeightVarsMemSize_; }
```
Comment (opti-mix) [inline include/glow/Backends/BackendUtils.h]: Same here.

File: include/glow/Backends/BackendUtils.h [hunk 10]
```diff
@@ -21,26 +21,65 @@
...
+  /// Type of symbol.
+  Type type;
+};
+
+/// Contains the information needed to be passed forward from compile time to
+/// runtime. In order to allocate and initialize memory.
+class RuntimeBundle {
+  /// Map from symbol name to a RuntimeSymbolInfo.
+  std::unordered_map<std::string, RuntimeSymbolInfo> symbolTable_;
+
```
Comment (opti-mix) [inline include/glow/Backends/BackendUtils.h]: Why the newline here?

File: include/glow/Backends/BackendUtils.h [hunk 11]
```diff
@@ -21,26 +21,65 @@
...
+  size_t getValueOffset(const Named *v);
+  /// Helper function, gets symbol info for \p v.
+  RuntimeSymbolInfo getSymbolInfo(const Named *v);
+  /// At compile time condense constants to a single block of memory.
+  /// This allows the graph to go away after compile time.
+  /// Allocates a block of memory of size \p constantMaxSize then walks the
+  /// given function \p F and and copies weights to their address as specified
+  /// by offsets contained in symbolTable_.
+  void collectConstants(const IRFunction *F);
+  RuntimeBundle(std::unordered_map<std::string, RuntimeSymbolInfo> symbolTable,
```
Comment (opti-mix) [inline include/glow/Backends/BackendUtils.h]: Should the `symbolTable` argument be passed by reference?
